programs/bpf_loader/src/syscalls.rs

```diff
@@ -2717,6 +2734,80 @@ impl<'a> SyscallObject<BpfError> for SyscallGetReturnData<'a> {
     }
 }
 
+// Event handling
+pub struct SyscallEvent<'a> {
+    invoke_context: Rc<RefCell<&'a mut dyn InvokeContext>>,
+    loader_id: &'a Pubkey,
+}
+impl<'a> SyscallObject<BpfError> for SyscallEvent<'a> {
+    fn call(
+        &mut self,
+        addr: u64,
+        len: u64,
+        _arg3: u64,
+        _arg4: u64,
+        _arg5: u64,
+        memory_mapping: &MemoryMapping,
+        result: &mut Result<u64, EbpfError<BpfError>>,
+    ) {
+        let invoke_context = question_mark!(
+            self.invoke_context
+                .try_borrow()
+                .map_err(|_| SyscallError::InvokeContextBorrowFailed),
+            result
+        );
+
+        let budget = invoke_context.get_compute_budget();
+
+        question_mark!(
```

Comment:
Can't we go ahead and consume `len` up front?

Reply:
We need to get the sum of all of the length of all the fields.

Reply:
True dat

Reply:
I've used the same costing model as `sol_log`